### ISSUE TITLE
add new nushell syntax for input output types

### ISSUE THREAD

**Issue by fdncred** [nushell/vscode-nushell-lang#159]
JT added some new syntax to custom commands to allow one to specify the input and output types. You can see the details in [PR9690](https://github.com/nushell/nushell/pull/9690)

For example (a single int-to-int input/output pair):

```nushell
def foo []: int -> int { ... }
```

You can also have multiple input/output pairs:
```nushell
def bar []: [int -> string, string -> list<string>] { ... }
```

@glcraft I hope it's not a problem tagging you on new syntax. You're help would be appreciated here too.

Related to https://github.com/nushell/vscode-nushell-lang/issues/146

**Comment by glcraft** [comment]
Yep ok, I'm going to check this out 👍

**Comment by fdncred** [comment]
closed by #160
